lukemckinstry/ebus-breakable-toy
# AWS deployment with terraform and supporting scripts

This PR adds:
* `infra` script and `/terraform` code directory to create AWS infrastructure for the project. The infrastructure is Elastic Container Service (ECS) on Fargate, includes a RDS service for the Postgres/PostGIS database, and sits in a VPC with 2 private and 2 public subnets in 2 availability zones, and includes a load balancer.  
* `cipublish` script to build container image and publish it to AWS Elastic Container Registry (ECR)
* `deploy_ecs.py` script to update ECS service with latest image in ECR

Testing:
At this stage of the project I do not want to leave the site or any AWS infrastructure (besides ECR) running for cost reasons. So the plan for now is to spin up the infrastructure as needed to test the application as development of the breakable toy continues. To test creating and using the infrastructure: 
* Build the AWS infrastructure by following the steps to use terraform code in `terraform/README.md`
* Access the running the container and ensure you can run database migrations or management commands using ECS Exec 
```bash
$ aws ecs execute-command --region us-east-1 --cluster app --task <ENTER-TASK-ID-HERE> --container sun-api --command "/bin/bash" --interactive
```
* Deploy changes to the infrastructure using `./scripts/cipublish` and `python update_ecs.py --service app --cluster sun_api`
